src/util/bitvector_expr.h

```diff
@@ -677,4 +677,146 @@ inline popcount_exprt &to_popcount_expr(exprt &expr)
   return ret;
 }
 
+/// \brief A Boolean expression returning true, iff operation \c kind would
+/// result in an overflow when applied to operands \c lhs and \c rhs.
+class binary_overflow_exprt : public binary_exprt
+{
```

Comment:
You could make it a `binary_predicate_exprt`

Reply:
Done.